drivers/include/bmp180.h

```diff
@@ -32,112 +32,107 @@ extern "C" {
  * @name Oversampling modes
  * @{
  */
-#define BMP180_ULTRALOWPOWER          (0)
-#define BMP180_STANDARD               (1)
-#define BMP180_HIGHRES                (2)
-#define BMP180_ULTRAHIGHRES           (3)
+typedef enum {
+    BMP180_ULTRALOWPOWER = 0,
+    BMP180_STANDARD,
+    BMP180_HIGHRES,
+    BMP180_ULTRAHIGHRES
+} bmp180_oversampling_mode_t;
 /** @} */
 
 /**
  * @brief Calibration struct for the BMP180 sensor
  */
 typedef struct {
-    int16_t ac1;          /**< ac1 coefficient */
-    int16_t ac2;          /**< ac2 coefficient */
-    int16_t ac3;          /**< ac3 coefficient */
-    int16_t b1;           /**< b1 coefficient */
-    int16_t b2;           /**< b2 coefficient */
-    int16_t mb;           /**< mb coefficient */
-    int16_t mc;           /**< mc coefficient */
-    int16_t md;           /**< md coefficient */
-    uint16_t ac4;         /**< ac4 coefficient */
-    uint16_t ac5;         /**< ac5 coefficient */
-    uint16_t ac6;         /**< ac6 coefficient */
+    int16_t ac1;                                /**< ac1 coefficient */
+    int16_t ac2;                                /**< ac2 coefficient */
+    int16_t ac3;                                /**< ac3 coefficient */
+    int16_t b1;                                 /**< b1 coefficient */
+    int16_t b2;                                 /**< b2 coefficient */
+    int16_t mb;                                 /**< mb coefficient */
+    int16_t mc;                                 /**< mc coefficient */
+    int16_t md;                                 /**< md coefficient */
+    uint16_t ac4;                               /**< ac4 coefficient */
+    uint16_t ac5;                               /**< ac5 coefficient */
+    uint16_t ac6;                               /**< ac6 coefficient */
 } bmp180_calibration_t;
 
 /**
- * @brief Device descriptor for the BMP180 sensor
+ * @brief Device initialization parameters
  */
 typedef struct {
-    i2c_t i2c_dev;                     /**< I2C device which is used */
-    bmp180_calibration_t calibration;  /**< Device calibration */
-    uint8_t oversampling;              /**< Oversampling mode */
-} bmp180_t;
-
+    i2c_t i2c_dev;                              /**< I2C device which is used */
+    bmp180_oversampling_mode_t oversampling;    /**< Oversampling mode */
+} bmp180_params_t;
 
 /**
- * @brief Device initialization parameters
+ * @brief Device descriptor for the BMP180 sensor
  */
 typedef struct {
-    i2c_t i2c_dev;   /**< I2C device which is used */
-    uint8_t mode;    /**< Oversampling mode */
-} bmp180_params_t;
+    bmp180_params_t      params;                /**< Device initialization parameters */
+    bmp180_calibration_t calibration;           /**< Device calibration */
```

Comment:
I'm actually asking, I haven't thought this through - maybe @PeterKietzmann or @haukepetersen can provide their 2 cents here, too?